### PR TITLE
[DCS-153] Geolocation

### PR DESCRIPTION
**Pull request by eruizgar**
## Description

- This feature extends Hermes with the porpouse of generate random geolocations. 
- The complete description of how use methods is in README.md included.

### Testing
- You could find unit test in HermesTest.scala
- [X] Unit, integration, acceptance tests as appropriate
- [X] Coverage (>90%)

### Documentation
- [ ] Changelog update
- [X] Documentation entry

### Scalastyle
Result of scalastyle execution:

### CODE REVIEW

File: src/main/resources/locales/geo/US.json [hunk 1]
```diff
@@ -0,0 +1,1506 @@
+[{"latitude": 39.6335556, "longitude": -86.8138056},
```
Comment (anistal) [inline src/main/resources/locales/geo/US.json]: Only 50 entries please

**Review by anistal — COMMENTED** [review]
All OK. You have forgotten to delete lines in US locale (that has tons of lines)

File: src/test/scala/com/stratio/hermes/utils/HermesTest.scala [hunk 2]
```diff
@@ -135,6 +135,33 @@ class HermesTest extends FlatSpec with Matchers {
     //scalastyle:on
   }
 
+  it should "generate valid locations: ES and US locales" in {
+
+    val hermesES = Hermes("ES")
+    hermesES.Geo.geoModel should contain (hermesES.Geo geolocation)
```
Comment (albertostratio) [inline src/test/scala/com/stratio/hermes/utils/HermesTest.scala]: It doesn't matter but to be consistent with the next check I'd add a period after the Geo -> (hermesEs.Geo.geolocation)

File: src/test/scala/com/stratio/hermes/utils/HermesTest.scala [hunk 3]
```diff
@@ -135,6 +135,33 @@ class HermesTest extends FlatSpec with Matchers {
...
+
+    val hermesUS = Hermes("US")
+    hermesUS.Geo.geoModel should contain (hermesUS.Geo.geolocation)
+  }
+
+  it should "raise an exception when it gets a geolocation are empty in the locale" in {
```
Comment (albertostratio) [inline src/test/scala/com/stratio/hermes/utils/HermesTest.scala]: The test descripcion is not clear I'd change it to be: it should "raise a NoSuchElementException when the locale is empty"

**Review by albertostratio — CHANGES_REQUESTED** [review]
@eruizgar I missed tests cases for some of our acceptance criteria: the ones where some records of our files have records corrupted, could you please add tests for these cases?

File: src/test/scala/com/stratio/hermes/utils/HermesTest.scala [hunk 4]
```diff
@@ -135,6 +135,41 @@ class HermesTest extends FlatSpec with Matchers {
...
+  }
+
+  it should "raise an exception when it gets a geolocation that is corrupted" in {
+    val hermesYY = Hermes("YY")
+    //scalastyle:off
+    hermesYY.Geo.parseErrorList(hermesYY.Geo.geoModel).length should be(1)
+    an[NoSuchElementException] should be thrownBy hermesYY.Geo.geolocation
+    //scalastyle:on
+  }
   /**
```
Comment (albertostratio) [inline src/test/scala/com/stratio/hermes/utils/HermesTest.scala:180]: With the changes that you've just made I guess we should include more tests to fulfill the acceptance criteria.

File: src/main/scala/com/stratio/hermes/utils/Hermes.scala [hunk 5]
```diff
@@ -151,8 +159,49 @@ case class Hermes(locale: String = HermesConstants.ConstantDefaultLocale) extend
...
+
+    /**
+     * Example: "geolocation() -> 57.585393,-157.571944".
+     * @return a random geolocation.
+     */
+    def geolocation(): (GeoModel) = {
+      RandomHelper.randomElementFromAList[(GeoModel)](geolocationOfGeoModel(geoModel)).getOrElse(throw new NoSuchElementException)
+    }
+
+    def geolocationOfGeoModel(l: List[Either[String, Seq[GeoModel]]]): List[GeoModel] = {
```
Comment (albertostratio) [inline src/main/scala/com/stratio/hermes/utils/Hermes.scala]: I would call this method: geoModelList

**Review by albertostratio — CHANGES_REQUESTED** [review]

**Review by albertostratio — APPROVED** [review]

**Comment by albertostratio** [comment]
Coverage = 95%, good joob @eruizgar !!

LGTM

**Pull request merged**